src/microbit/test/test_microbit_model.py

```diff
@@ -20,7 +21,21 @@ def test_running_time(self):
         mock_end_time = 300
         self.__mb._MicrobitModel__start_time = mock_start_time
         time.time = mock.MagicMock(return_value=mock_end_time)
-        print(time.time())
         assert mock_end_time - mock_start_time == pytest.approx(
             self.__mb.running_time()
         )
+
+    @pytest.mark.parametrize(
+        "temperature", [CONSTANTS.MIN_TEMPERATURE, 0, CONSTANTS.MAX_TEMPERATURE],
+    )
+    def test_temperature(self, temperature):
+        self.__mb._MicrobitModel__set_temperature(temperature)
+        assert temperature == self.__mb.temperature()
+
+    @pytest.mark.parametrize(
+        "invalid_temperature",
+        [CONSTANTS.MIN_TEMPERATURE - 10, CONSTANTS.MAX_TEMPERATURE + 5],

```

Comment:
_&ast;nit&ast;_
I would do `CONSTANTS.MIN_TEMPERATURE - 1` and `CONSTANTS.MAX_TEMPERATURE + 1` to check the corner cases instead.

Reply:
will change then merge! Thanks David